### DISCUSSION TITLE
_cfv - minor issue (?)

### DISCUSSION THREAD

**Discussion by Deanobeano2** [cortexrd/Knack-Toolkit-Library#129] [General]
Hi

I'm just starting and learning how to use the _cfv keyboard and seem to have hit a minor issue. I'm using _cfv =[field_356, is, Inactive, red, bold] and it correctly uses red for the text for the target field but I can't seem to get the bold to work. Am i doing something wrong or could there be something else preventing the target text from being in bold?

Dean

**Comment by CSWinnall** [comment]
Hi @Deanobeano2 

The _cfv parameters need to be in a certain order you have put the bold in the background-color slot.

Try this:

_cfv=[field_356, is, Inactive, red, , bold, ] 

The cfv has 7 parameters which all need to be in a certain order for it to work.

Craig

**Comment by Deanobeano2** [comment]
Craig

Thank you very much for the prompt reply. I should have paid more attention to the (required) in the parameters, and the ,,, in some examples I'd seen.

That quickly solved that problem and has enabled me to make progress on the next challenge of trying to hide a field in a form contained within a scene, it that is even/currently possible.

Dean

**Reply by CSWinnall** [reply]
Hi Dean

If you just want to hide it then use _hf.  If you need it to be conditional then use cfv with the last parameter (options) being ,h

If it is neither of these things can you give an example please?

Craig

**Comment by Deanobeano2** [comment]
Hi Craig

I have a scene for sales order entry comprised of the Sales Order at the top, a grid of the line items for the sales order in the middle, and a form at the bottom for adding items to the order. In this instance I'd like like to hide the Location field in the form if the location field in the Sales Order is not blank, as we'll use record rules to populate the location field for the line item.

I hope this helps describe what I'm trying to achieve.

Dean

**Comment by CSWinnall** [comment]
The first form is that submitted before the second form?

Is the location added in the first form when the scene is opened or is it
already populated?

There is a certain set of circumstances where cfv may work but I think this
might be a JQuery coding exercise.

If you could provide a screenshot that would be helpful and whether each
form is an add or an update form, please.

Craig

**Comment by Deanobeano2** [comment]
Hi Craig

Hopefully, the attached pictures help you understand our/the process.
![Add Order](https://github.com/cortexrd/Knack-Toolkit-Library/assets/130625963/1ef29613-87d3-4090-8fe5-24178b27763f)
![Order Details, Line Items, and Add Line Item form](https://github.com/cortexrd/Knack-Toolkit-Library/assets/130625963/ad49adaf-46b1-4f7e-8bd5-b01b898b08dc)

Dean

**Comment by CSWinnall** [comment]
Hi Dean

I don't think this is possible using just KTL (yet) it will require some code.  I don't mind helping with that if you'd like?

Craig

**Comment by Deanobeano2** [comment]
Hi Craig

Norm thought this probably wasn't possible from an earlier description, and with more information you've confirmed our suspicions.

It would be appreciated if the solution could be achieved via code but only if it wouldn't take too long, and you could see the code being of benefit to other people. Otherwise, it might be best to wait until this feature is added as I think Norm said it might be down the track.

Dean

**Reply by CSWinnall** [reply]
Hi Dean

It looks like the Norm has added ktlCond to the hide fields keyword this may be what you need:

`_hf=[field_xxx], [ktlCond, is, , field_yyy]`

If this doesn't work using the form field, you may need to add a details view with the location field of the sales order somewhere on the page use _hv to hide the view and use:

`_rvs=SalesOrderDetailsView`

On the Sales Order form.

Craig

**Reply by CSWinnall** [reply]
Sorry I have updated the _hf keyword so it uses the correct order of the ktlCond.

**Comment by cortexrd** [comment]
Guys,

Craig is right.  I've released a major update that will change everything, thanks to the `ktlCond` option.

I am updating the documentation right now.

More will follow tomorrow,
Norm

**Comment by amandajane-mo** [comment]
Hi Norm,

That’s gonna be amazing, looking forward to testing it out.

Thanks
Amanda

**Comment by Deanobeano2** [comment]
Hi Craig

Unfortunately, I can't seem to get it to work using your code nor with this variation:
_hf=[field_1510,view_353], [ktlCond, NOT, "", field_88,view_260]

Dean

**Comment by Deanobeano2** [comment]
Hi Craig

The issue with the code not working must be particular to this instance/view configuration as this test code:
 _hf=[field_1718], [ktlCond, IS, Lent, field_39]

used comparing values within the same view on another form works fine. I actually want the text for the field to be red and bold but thought I'd test things while I was playing around.

Dean

**Reply by CSWinnall** [reply]
Hi Dean

I need to take a look at Norms updates before I can offer advice on this as there have been a lot this weekend.  I am away until tomorrow then I'll take a good look and see if I can help.  

Craig

**Comment by Deanobeano2** [comment]
Hi Craig

That's quite all right as there's no urgency with this.

Dean
